tests/test-class-amp-validation-utils.php

```diff
@@ -68,16 +68,33 @@ class Test_AMP_Validation_Utils extends \WP_UnitTestCase {
 	 */
 	const TAG_NAME = 'img';
 
+	/**
+	 * Backed up $wp_registered_widgets.
+	 *
+	 * @var array
+	 */
+	protected $original_wp_registered_widgets;
+
 	/**
 	 * Setup.
 	 *
 	 * @inheritdoc
+	 * @global $wp_registered_widgets
 	 */
 	public function setUp() {
 		parent::setUp();
 		$dom_document = new DOMDocument( '1.0', 'utf-8' );
 		$this->node   = $dom_document->createElement( self::TAG_NAME );
 		AMP_Validation_Utils::reset_validation_results();
+		$this->original_wp_registered_widgets = $GLOBALS['wp_registered_widgets'];
+	}
+
+	/**
+	 * After a test method runs, reset any state in WordPress the test method might have changed.
+	 */
+	public function tearDown() {
+		$GLOBALS['wp_registered_widgets'] = $this->original_wp_registered_widgets; // WPCS: override ok.
```

Comment:
Good idea to reset the original global `$wp_registered_widgets`.

Reply:
It should be something that gets reset for us automatically, but it wasn't so this is to ensure we reset the global state.